src/Microsoft.AspNetCore.Blazor/Json/SimpleJson/SimpleJson.cs

```diff
@@ -1478,6 +1521,18 @@ public virtual object DeserializeObject(object value, Type type)
             return obj;
         }
 
+        internal virtual IDictionary<string, string> GetPascalCasePropertyNames(ICollection<string> keys)
+        {
+            var pascalCaseJsonPropertyKeyPair = new Dictionary<string, string>();
```

Comment:
Doing extra allocations for this isn't ideal.

On further investigation, it turns out that Json.NET does case-insensitive matches during deserialization. If it's good enough for Json.NET, then it's probably good enough for SimpleJson. I think we can accept the limitation that deserialization picks property names case-insensitively. Will this allow you to remove these changes that allocate a dictionary?

Reply:
Great. If possible please try to keep the overall code diff minimal, as we are still trying to manage this as a “fork” of SimpleJson, not an independent implementation. If the diff is too big and hard to reason about we might not be able to accept the change. Hope that’s ok.

Reply:
👍  Understood will keep that in mind.